docs/core/deploying/native-aot.md

```diff
@@ -0,0 +1,88 @@
+
+> [!WARNING]
+> Native AOT is supported in .NET 7 but only a limited number of libraries are fully compatible with native AOT in .NET 7.
+
+## Publish Native AOT - CLI
+
+Publish a native AOT application using the [dotnet publish](../tools/dotnet-publish.md) command.
+
+01. Add `<PublishAot>true</PublishAot>` to your project file.
+
+    This will produce a native AOT app and shows PublishAot compatibility warnings during build.
+
+    ```xml
+    <PropertyGroup>
+        <PublishAot>true</PublishAot>
+    </PropertyGroup>
+    ```
+
+02. Ensure pre-requisites that are needed for publishing a native AOT application are in the machine.
+
+    For Windows pre-requisites, install [Visual Studio 2022](https://visualstudio.microsoft.com/vs/), including Desktop development with C++ workload.
+
+    For Linux pre-requisites, install clang and developer packages for libraries that .NET runtime depends on.
+
+    *Ubuntu (18.04+)*
+
+    ```sh
+    sudo apt-get install clang zlib1g-dev
+    ```
+
+    *Alpine (3.15+)*
+
+    ```sh
+    sudo apk add clang gcc lld musl-dev build-base zlib-dev
+    ```
+
+03. Publish the app for a specific runtime identifier using `dotnet publish -r <RID>`.
+
+    The following example publishes the app for Windows as a native AOT application on a machine with the required pre-requisites installed.
+
+    `dotnet publish -r win-x64 -c Release`
+
+    The following example publishes the app for Linux as a native AOT application. A native AOT binary produced on Linux machine is only going to work on same or newer Linux version. For example, native AOT binary produced on Ubuntu 20.04 is going to run on Ubuntu 20.04 and later, but it is not going to run on Ubuntu 18.04.
+
+    `dotnet publish -r linux-arm64 -c Release`
+
+The app will be available in the publish directory and will contain all the code needed to run in it, including the coreclr runtime.
+
+## Impact of using the Native AOT deployment
+
+Native AOT applications are [self-contained](index.md#publish-self-contained) and don't use any shared libraries, as is the case with the default [framework-dependent deployment model](index.md#publish-framework-dependent). Consider the native AOT deployment model when the benefits to deployment and the ability to optimize a single app outweigh the benefits of sharing for certain types of workloads. The gain is most significant for workloads with a high number of deployed instances: cloud infrastructure, hyper-scale services, popular apps, or games.
+
+Native AOT applications comes with a few fundamental limitations and compatibility issues. The key limitations include:
+
+- No dynamic loading (for example, `Assembly.LoadFile`)
```

Comment:
I'm a little concerned with this message as is - native AOT will trim unused code and will only include the code that the app requires.

Reply:
They will still be bigger than FDD apps, right?

Reply:
I think SCD-FDD terminology might confuse with Native AOT with regards to talking about the size as well as the implication that the coreclr runtime is a separate entity. Native AOT is really a stand alone native binary (the closest would be to say its a trimmed SCD single file application with stripped down version of the runtime embedded in it but that would be confusing as well).

Reply:
I don't think that's the right frame. I grasp that people thinking working on Native AOT think of it as a separate universe. In order for us to think about it as supported and for it to be integrated into this doc-set, we need to think of it as another point on the spectrum of deployment choices.

The spectrum looks like this:

portable FDD -> RID-specific FDD -> self-contained -> single file -> native AOT

> Native AOT is really a stand alone native binary (the closest would be to say its a trimmed SCD single file application with stripped down version of the runtime embedded in it but that would be confusing as well).

That's thinking of it all as a series of tech choices. In reality, native aot sits naturally at the point I put it at in that spectrum.

Reply:
I like your framing native aot in terms of deployment choices like this but worry that native aot apps including the runtime libraries as a size limitation might not be clear to developers. The choices are really between shared vs non-shared library code but given that the doc briefly mentions the benefits of native aot apps at the start, I'm ok with FDD-SCD phrasing.

If you can think of a way to word this better, please do so. Otherwise I'll commit the suggestion above.

Reply:
> , as compared to framework-dependent apps

I added that. Does that help?

We could also add it last in the list. That might help.